File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        var freq = new Dictionary<int, int>();

        foreach(var i in nums) {
            if (freq.TryGetValue(i, out var v)) {
                freq[i] = v + 1;
            } else {
                freq[i] = 1;
            }
        }

        var q = new PriorityQueue<int, int>();
        
        foreach (var (val,f) in freq) {
            q.Enqueue(val,-f);
        }


        var result = new int[k];

        for (int i = 0; i < k; i++) {
            result[i] = q.Dequeue();
        }

        return result;        
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.go
func longestConsecutive(nums []int) int {
	set := map[int]bool{}

	for _, n := range nums {
		set[n] = true
	}

	result := 0

	for _, n := range nums {
		if _, e:= set[n-1]; !e {
			length := 1
			for range len(nums) {
				if _, nE := set[n+length]; nE {
					length++
				} else {
					break
				}
			}

			if length > result {
				result = length
			}
		}
	}

	return result
}
